api/maven-api-core/src/main/java/org/apache/maven/api/services/ProjectBuilder.java

```diff
@@ -0,0 +1,57 @@
+package org.apache.maven.api.services;
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import org.apache.maven.api.annotations.Nonnull;
+
+import java.nio.file.Path;
+
+import org.apache.maven.api.Session;
+import org.apache.maven.api.Artifact;
+
+public interface ProjectBuilder extends Service
+{
+
+    @Nonnull
+    ProjectBuilderResult build( ProjectBuilderRequest request )
+        throws ProjectBuilderException, IllegalArgumentException;
+
+    @Nonnull
+    default ProjectBuilderResult build( @Nonnull Session session, @Nonnull ProjectBuilderSource source )
+            throws ProjectBuilderException, IllegalArgumentException
+    {
+        return build( ProjectBuilderRequest.build( session, source ) );
+    }
+
+    @Nonnull
+    default ProjectBuilderResult build( @Nonnull Session session, @Nonnull Path path )
+            throws ProjectBuilderException, IllegalArgumentException
```

Comment:
Don't use runtime exception `IllegalArgumentException` in method signature.

Reply:
You can describe it in Javadoc but don't use it in method signature. See the JDK, this is a real case.